### PR TITLE
[FIX] account_reversal: allow reversing move in closed periods, fixes #265

### LINKED ISSUE #265: account_reversal: cannot reverse a move that is in a closed period
How to reproduce:
- create move M in period P
- close period P
- try to reverse move M in period P+1
- => You can not add/modify entries in a closed period P of journal ...

This is due to https://github.com/OCA/account-financial-tools/blob/8.0/account_reversal/account_reversal.py#L88 which tries to write on the original move.

### PR DESCRIPTION
**Pull request by sbidoul**
Fixes #265

### CODE REVIEW

File: account_reversal/account_reversal.py [hunk 1]
```diff
@@ -40,6 +40,14 @@ class account_move(models.Model):
         readonly=True)
 
     @api.multi
+    def validate(self):
+        # TODO: remove this method if and when
+        #       https://github.com/odoo/odoo/pull/7735 is merged
+        if self.env.context.get('novalidate'):
+            return
+        super(account_move, self).validate()
```
Comment (adrienpeiffer) [inline account_reversal/account_reversal.py]: You should return the value

Comment (sbidoul) [inline account_reversal/account_reversal.py]: @adrienpeiffer done. Well spotted.

**Comment by adrienpeiffer** [comment]
:+1: code review and functionnal test locally with #267

File: account_reversal/tests/test_account_reversal.py [hunk 2]
```diff
@@ -34,12 +34,10 @@ def setUp(self):
...
         self.move_line_obj = self.env['account.move.line']
 
-    def _create_move(self, with_partner, amount=100):
+    def _create_move(self, with_partner, amount=100, period=None):
         date = datetime.now()
         company_id = self.env.ref('base.main_company').id
-        period_id = self.env['account.period'].with_context(
-            account_period_prefer_normal=True,
-            company_id=self.env.ref('base.main_company').id).find(date)[0]
+        period = period or self.env.ref('account.period_0')
```
Comment (pedrobaeza) [inline account_reversal/tests/test_account_reversal.py:40]: Are you making now the test with fixed period data?

Comment (sbidoul) [inline account_reversal/tests/test_account_reversal.py:40]: @pedrobaeza I don't understand your comment, can you elaborate?

Comment (pedrobaeza) [inline account_reversal/tests/test_account_reversal.py:40]: Nevermind, I see it later that it's a fallback value

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**